nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,74 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  try {
+    console.log("in/api/meals");
+    // variable for all the querys I am using so that I can check if one of the requested queryes from the user doesn't match/exists
+    const querys = ["maxPrice", "title", "createdAfter", "limit"];
+    const requestQuerys = Object.keys(request.query);
+    const matchQuerys = [];
+    requestQuerys.some((item) => {
+      if (querys.includes(item)) {
+        matchQuerys.push(item);
+      }
+    });
+    // new array with let, that contain all the meals, to reasign values
+    let sortedMeals = meals;
+    const { maxPrice, title, createdAfter, limit } = request.query;
+    
+    if (matchQuerys.length === requestQuerys.length) {
+      // maxPrice	Get meals that has a price smaller than maxPrice	Number	/api/meals?maxPrice=90
+      if (maxPrice) {
+        sortedMeals = sortedMeals.filter(
+          (meal) => meal.price < parseInt(maxPrice)
+          );
+        }
+        // title	Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde	String	/api/meals?title=Indian%20platter
+        if (title) {
+          const decodeTitle = decodeURI(title).toLowerCase();
+          sortedMeals = sortedMeals.filter((meal) =>
+          meal.title.toLowerCase().includes(decodeTitle)
```

Comment:
Nice! When comparing strings it's generally a good practice to first convert them to lowercase, well done!